exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/DrillCompoundIdentifier.java

```diff
@@ -69,31 +70,38 @@ public void addIndex(int index, SqlParserPos pos){
     }
   }
 
-  public SqlNode getAsSqlNode(){
-    if(ids.size() == 1){
+  public SqlNode getAsSqlNode(Set<DrillCompoundIdentifier> fullSchemasSet) {
```

Comment:
I can't argue with the facts.

But you're writing an ugly piece of code and building up technical debt. That is a bad idea.

Maybe you need to revisit how you deal with JSON arrays.